src/PAModel/CanvasDocument.cs

```diff
@@ -625,16 +623,19 @@ private int FindMaxEntropyFileName()
         private void RestoreAssetFilePaths()
         {
             // For apps unpacked before this asset rewrite was added, skip the restore step
-            if (_entropy.LocalResourceFileNames == null)
+            if (_entropy?.LocalResourceFileNames == null)
```

Comment:
You would still need to separate it out `if (_entropy == null || _entropy.LocalResourceFileNames == null)`

Reply:
I think null coalescing handles both scenarios. I think _entropy?.LocalResourceFileNames will be resolved to null by   short circuiting to null when _entropy itself is null or yielding null as the final result if _entropy is not null, but LocalResourceFileNames is null. If the preferred style is to break it down into two separate comparisons, then I can break the expression down and update the pr accordingly.